packages/babel-helper-create-class-features-plugin/src/misc.js

```diff
@@ -72,8 +72,14 @@ export function injectInitialization(path, constructor, nodes, renamer) {
   if (isDerived) {
     const bareSupers = [];
     constructor.traverse(findBareSupers, bareSupers);
+    let isFirst = true;
     for (const bareSuper of bareSupers) {
-      bareSuper.insertAfter(nodes);
+      if (isFirst) {
```

Comment:
The test [`legacy-class-prototype-properties/child-classes-properties`](https://github.com/babel/babel/blob/v7.10.4/packages/babel-plugin-proposal-decorators/test/fixtures/legacy-class-prototype-properties/child-classes-properties/exec.js) is [failing](https://github.com/JLHwung/babel/runs/851654026#step:4:767) if we change `bareSuper.insertAfter(nodes.map(n => t.cloneNode(n)))` unconditionally. The current approach happens to make our test cases happy and also slightly faster because we get rid of redundant clone.

However, I think it reveals an issue which is not caught in our current test cases.

I suggest we address that in a separate PR or even let it stay in the backlog since it is related to legacy decorators only.

I am not familiar with decorators part, suggestions are definitely welcome.

Reply:
It is probably failing because in the decorators plugin we use a `WARNING_CALLS` WeakSet to track warnings: if we clone the node we cannot find it anymore in the set.